DirectX11Proj/ResourceManager.cpp

```diff
@@ -1,29 +1,65 @@
 #include "ResourceManager.h"
 
+#include <string>
+#include <iostream>
+#include <memory.h>
+
 #include "modelclass.h"
 #include "ModelData.h"
+#include "d3dTexture.h"
 
 #include "easylogging++.h"
-
 ResourceManager::ResourceManager()
 {
 	mpModelLoader = std::make_unique<ModelLoader>();
 }
 
+void ResourceManager::Shutdown()
+{
+	this->mLoadedModels.clear();
+	this->mLoadedTextures.clear();
+}
+
 
 // TODO, create a function for model loader which just asks for the next model, this removes it from the model list
 std::vector<ModelClass*> ResourceManager::LoadModels(std::string aFilePath)
 {
 	mpModelLoader->LoadModel(aFilePath.c_str());
 	std::vector<ModelClass*> tModels;
 
-	std::shared_ptr<ModelClass> tModelClass = nullptr;
-	
-	for (unsigned int i = 0; i < mpModelLoader->GetMeshesToBeProcessed().size(); ++i)
+	std::unique_ptr<ModelClass> tModelClass = nullptr;
+	std::unique_ptr<d3dTexture> tTextureClass = nullptr;
+
+	const std::vector<MeshData>& tMeshes = mpModelLoader->GetMeshesToBeProcessed();
+
+	for (unsigned int i = 0; i < tMeshes.size(); ++i)
 	{
-		tModelClass = std::make_shared<ModelClass>();
+		const MeshData& tData = tMeshes[i];
+
+		tModelClass = std::make_unique<ModelClass>();
 		tModelClass->Initialize(this->mpDevice, mpModelLoader->GetMeshesToBeProcessed()[i]);
 
+		tTextureClass = std::make_unique<d3dTexture>();
+
+		// Convert string texture filepath to wstring
+	
+		std::wstring wString = std::wstring(tData.textureFilePath.begin(), tData.textureFilePath.end());
+		const WCHAR* result = wString.c_str();
+
+		// if the single textuer could not initialize
+		if (!tTextureClass->Initialize(mpDevice, result))
+		{
+			LOG(WARNING) << "ModelLoading: Finished loading model " << aFilePath;
+		}
+
```

Comment:
may want to remove the enter between the if and else, so you know they're both related to each other and no accidental code can get in between